src/main/java/backend/LoginController.java

```diff
@@ -61,7 +61,8 @@ public void getPreviousLoginDetails() {
 
     public boolean attemptLogin() {
 
-        boolean validRepoId = Utility.isWellFormedRepoId(owner, repo) && !username.isEmpty() && !password.isEmpty();
+        boolean validRepoId = RepoId.isWellFormedRepoIdString(owner + "/" + repo)
+                && !username.isEmpty() && !password.isEmpty();
         boolean loginSuccessful = login(username, password);
 
         if (validRepoId && loginSuccessful) {
```

Comment:
@garbanzos is right about what the check essentially does.

That method is doing multiple things, and probably should only be concerned with supplying the login details to EGit. We can factor out the validation logic so it's checked before this point.